### PR TITLE
Issue 183 & documentation

### PR DESCRIPTION
**Pull request by kdorheim**
Minor PR that fixes some typos and adds a date to the csv output stream as requested in #183

### CODE REVIEW

**Review by bpbond — COMMENTED** [review]
Um...a PR that will break _every single analysis script that reads the primary Hector output file_ is not minor! 😄 

@kdorheim Let's discuss this?

File: src/temperature_component.cpp [hunk 1]
```diff
@@ -192,7 +192,7 @@ void TemperatureComponent::setData( const string& varName,
              lo_warming_ratio = data.getUnitval(U_UNITLESS);
         } else {
             H_THROW( "Unknown variable name while parsing " + getComponentName() + ": "
-                    + varName );
+                                                                                                   );
```
[the remark this answers is not in the document]
Comment (kdorheim) [reply src/temperature_component.cpp:195]: umm no idk what this is, let me clean this up.

**Comment by kdorheim** [comment]
Oh 🙈 cause a new line is added? What if I added the date information to the first row `# Output from hector version 3.0`?

**Comment by bpbond** [comment]
Right, exactly. We can have a two-line header, or a header with an indeterminate number of lines but all prefixed with "#", but those are bigger changes. Happy to consider one of those options, but yeah, adding it to the end of the first line might be simpler.

**Review by bpbond — APPROVED** [review]

**Comment by bpbond** [comment]
@kdorheim 👍  feel free to merge when checks pass

**Comment by kdorheim** [comment]
Haha alright so with the latest commit, I believe that this PR is now minor documentation PR. 

Oh one question - In order to get the date information I did have to add a `#include <regex>` is that adding a dependency?

**Comment by bpbond** [comment]
No, `regex` is an included part of the standard C++ library. All good there.

**Comment by kdorheim** [comment]
Ughh okay to merge even though the vignettes are failing to build on mac OSX? Or should I look into it, they are passing on other platforms.

**Comment by bpbond** [comment]
Yes OK. The error message looks like a random one-off.

**Pull request merged**